components/brave_wallet/browser/solana_tx_manager_unittest.cc

```diff
@@ -795,4 +837,83 @@ TEST_F(SolanaTxManagerUnitTest, DropTxWithInvalidBlockhash) {
   EXPECT_EQ(pending_transactions[0]->id(), meta_id2);
 }
 
+TEST_F(SolanaTxManagerUnitTest, GetTransactionMessageToSign) {
+  // Unknown tx_meta_id yields null message
+  TestGetTransactionMessageToSign("Unknown", absl::nullopt);
+
+  const std::string from = "89DzXVKJ79xf9MkzTxatQESh5fcvsqBo9fCsbAXkCaZE";
+  const std::string to = "148FvZU6e67eSB12wv7fXCH5FsTDW8tsxXo3nFuZhfCF";
+  mojom::SolanaTxDataPtr system_transfer_data = nullptr;
+  TestMakeSystemProgramTransferTxData(from, to, 1, nullptr,
+                                      mojom::SolanaProviderError::kSuccess, "",
+                                      &system_transfer_data);
+  ASSERT_TRUE(system_transfer_data);
+  std::string system_transfer_meta_id;
+  AddUnapprovedTransaction(std::move(system_transfer_data), from,
+                           &system_transfer_meta_id);
+  ASSERT_FALSE(system_transfer_meta_id.empty());
+
+  // Invalid latest blockhash yields null message
+  SetInterceptor("", 0, "");
+  TestGetTransactionMessageToSign(system_transfer_meta_id, absl::nullopt);
+
+  // Valid latest blockhash yields valid transaction message to sign
+  SetInterceptor(latest_blockhash1_, last_valid_block_height1_, "");
+  absl::optional<std::vector<std::uint8_t>> message = base::Base64Decode(
```

Comment:
Can we add some tests for a null message and a `""` message here (I'm assuming those are different in C++ but not certain)?

Reply:
Yeah so the message is fetched when in [SolanaMessage::Serialize](https://github.com/brave/brave-core/blob/c900003c8f915063e357aa7339cdac978038ec19/components/brave_wallet/browser/solana_message.cc#L107), which can return `absl::nullopt` in a few cases - one of which is when `recent_blockhash_.empty()` which there's a case for

Reply:
👍🏻 thanks for that reference. I see that we do a check for the headers there as well so the `""` case would end up falling under the `absl::nullopt` case as well. This seems good to me then